src/hyperactive/integrations/sktime/_detector.py

```diff
@@ -0,0 +1,113 @@
+
+import numpy as np
+from skbase.utils.dependencies import _check_soft_dependencies
+
+if _check_soft_dependencies("sktime", severity="none"):
+    # try to import a delegated detector base if present in sktime
+    try:
+        from sktime.annotation._delegate import _DelegatedDetector
+    except Exception:
+        from skbase.base import BaseEstimator as _DelegatedDetector
+else:
+    from skbase.base import BaseEstimator as _DelegatedDetector
+
+from hyperactive.experiment.integrations.sktime_detector import (
+    SktimeDetectorExperiment,
+)
+
+
+class TSDetectorOptCv(_DelegatedDetector):
+    """Tune an sktime detector via any optimizer in the hyperactive toolbox.
+
+    This mirrors the interface of other sktime wrappers in this package and
+    delegates the tuning work to `SktimeDetectorExperiment`.
+    """
+
+    _tags = {
+        "authors": "fkiraly",
+        "maintainers": "fkiraly",
+        "python_dependencies": "sktime",
+    }
+
+    _delegate_name = "best_detector_"
+
+    def __init__(
+        self,
+        detector,
+        optimizer,
+        cv=None,
+        scoring=None,
+        refit=True,
+        error_score=np.nan,
+        backend=None,
+        backend_params=None,
+    ):
+        self.detector = detector
+        self.optimizer = optimizer
+        self.cv = cv
+        self.scoring = scoring
+        self.refit = refit
+        self.error_score = error_score
+        self.backend = backend
+        self.backend_params = backend_params
+        super().__init__()
+
+    def _fit(self, X, y):
+        detector = self.detector.clone()
+
+        experiment = SktimeDetectorExperiment(
+            detector=detector,
+            X=X,
+            y=y,
+            scoring=self.scoring,
+            cv=self.cv,
+            error_score=self.error_score,
+            backend=self.backend,
+            backend_params=self.backend_params,
+        )
+
+        optimizer = self.optimizer.clone()
+        optimizer.set_params(experiment=experiment)
+        best_params = optimizer.solve()
+
+        self.best_params_ = best_params
+        self.best_detector_ = detector.set_params(**best_params)
+
+        if self.refit:
+            # detectors typically implement fit(X, y) or fit(X)
+            try:
+                self.best_detector_.fit(X=X, y=y)
+            except TypeError:
+                self.best_detector_.fit(X=X)
+
+        return self
+
+    def _predict(self, X):
+        if not self.refit:
+            raise RuntimeError(
+                f"In {self.__class__.__name__}, refit must be True to make predictions,"
+                f" but found refit=False. If refit=False, {self.__class__.__name__} can"
+                " be used only to tune hyper-parameters, as a parameter estimator."
+            )
+        return super()._predict(X=X)
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        # Import sktime DummyDetector only if sktime is available; otherwise
```

Comment:
* use `_check_soft_dependencies` to detect if `sktime` is available
* add a second test parameter set, where you set more parameters